lib/gulp-tasks/patterns-import.js

```diff
@@ -0,0 +1,72 @@
+/**
+ *  @fileOverview
+ * Uses Gulpjs to import html patterns
+ *
+ *  @author       Scott Nath
+ *
+ *  @requires     NPM:gulp
+ *  @requires     NPM:lodash.merge
+ *  @requires     NPM:pattern-importer
+ */
+
+var gulp = require('gulp'),
+  merge = require('lodash.merge'),
+  utility = require('../utility');
+
+/**
+ * Function to get default options for an implementation of patternlab-import
+ * @returns {Object}  options  an object of default patternlab-import options
+ */
+var getDefaultOptions = function () {
+  'use strict';
+  /* default options for angularTemplatecache gulp task */
+  var options = {
+    config: {
+      compilePatternsOnImport: false,
+      dataSource: 'pattern',
+      dataFileName: 'pattern.yml',
+      htmlTemplateDest: './source/_patterns',
+      stylesDest: './source/css/scss',
+      scriptsDest: './source/js',
+      cssCompiler: 'sass', // sass, less, stylus, none
+      templateEngine: 'twig',
+      templateEngineOptions: {
+        'base': 'node_modules/pattern-library/patterns/',
+        'async': false
+      },
+      templateDonut: {
+        'twig': './node_modules/pattern-importer/templates/donut.twig'
+      },
+      convertCategoryTitles: true
+    },
+    src: ['./node_modules/pattern-library/patterns/**/pattern.yml'],
+    taskName: 'patterns-import', // default task name
+    dependencies: [] // gulp tasks which should be run before this task
+  };
+
+  return options;
+}
+
+/**
+ * Gulp task to import raw patterns and convert them to browser-ready html/css/js
+ * @name patternsImport
+ * @param {Object} options custom options
+ * @param {Array|String} options.patternFiles  project-relative path to sets of un-compiled patterns
+ * @param {Object}  options.patternImporterOptions  options needed for the pattern-importer
+ * @requires NPM:Gulp
+ * @requires NPM:lodash.merge
+ */
+exports.gulpImportPatterns = function (gulp, projectOptions) {
```

Comment:
each gulp task inside this repository is here so that it can be used by requiring `pattern-library-utilities` as a node module in other modules. Then we just need to inject a configuration and require the task and it's ready to us in your projects.

So, for this task, it's used by the gulpfile for any pattern library created with generator-pattern-library.
See:
https://github.com/pattern-library/generator-pattern-library/blob/master/generators/app/templates/_gulpfile.js#L83